redbot/cogs/admin/admin.py

```diff
@@ -450,45 +450,70 @@ async def selfroleset(self, ctx: commands.Context):
-            await ctx.send(
-                _(
-                    "I cannot let you add {role.name} as a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
-                ).format(role=role)
-            )
-            return
-        async with self.config.guild(ctx.guild).selfroles() as curr_selfroles:
-            if role.id not in curr_selfroles:
-                curr_selfroles.append(role.id)
-                await ctx.send(_("Added."))
+        count = 0
+        current_selfroles = await self.config.guild(ctx.guild).selfroles()
+        for role in roles:
+            if not self.pass_user_hierarchy_check(ctx, role):
+                await ctx.send(
+                    _(
+                        "I cannot let you add {role.name} as a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
+                    ).format(role=role)
+                )
+                return
+            if role.id not in current_selfroles:
+                current_selfroles.append(role.id)
+                count += 1
+            else:
+                await ctx.send(
+                    _('The role "{role.name}" is already a selfrole.').format(role=role)
+                )
                 return
 
-        await ctx.send(_("That role is already a selfrole."))
+        await self.config.guild(ctx.guild).selfroles.set(current_selfroles)
+        await ctx.send(
+            _("Added {count} {plural_roles}.").format(
+                count=str(count), plural_roles="selfroles" if count > 1 else "selfrole"
+            )
+        )
 
     @selfroleset.command(name="remove")
-    async def selfroleset_remove(self, ctx: commands.Context, *, role: SelfRole):
+    async def selfroleset_remove(self, ctx: commands.Context, *roles: SelfRole):
         """
-        Remove a role from the list of available selfroles.
+        Remove a role, or a selection of roles, from the list of available selfroles.
 
         NOTE: The role is case sensitive!
         """
-        if not self.pass_user_hierarchy_check(ctx, role):
-            await ctx.send(
-                _(
-                    "I cannot let you remove {role.name} from being a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
-                ).format(role=role)
-            )
-            return
-        async with self.config.guild(ctx.guild).selfroles() as curr_selfroles:
-            curr_selfroles.remove(role.id)
+        count = 0
+        current_selfroles = await self.config.guild(ctx.guild).selfroles()
+        for role in roles:
+            if not self.pass_user_hierarchy_check(ctx, role):
+                await ctx.send(
+                    _(
+                        "I cannot let you remove {role.name} from being a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
```

Comment:
Styling
```suggestion
                        "I cannot let you remove {role.name} from being a selfrole"
                        " because that role is higher than or equal to your highest role"
                        " in the Discord hierarchy."
```